src/test/java/com/karankumar/bookproject/backend/service/AuthorServiceTest.java

```diff
@@ -0,0 +1,81 @@
+package com.karankumar.bookproject.backend.service;
+
+import com.karankumar.bookproject.annotations.IntegrationTest;
+import com.karankumar.bookproject.backend.entity.Author;
+import com.karankumar.bookproject.backend.repository.AuthorRepository;
+import lombok.extern.java.Log;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@Log
+@IntegrationTest
+public class AuthorServiceTest {
+
+    private final AuthorService sut;
+    private final BookService bookService;
+    private final AuthorRepository authorRepository;
+
+    @Autowired
+    public AuthorServiceTest(AuthorService sut, BookService bookService, AuthorRepository authorRepository) {
+
+        this.sut = sut;
+        this.bookService = bookService;
+        this.authorRepository = authorRepository;
+    }
+
+    @BeforeEach
+    void teardown() {
+        bookService.deleteAll();
+        authorRepository.deleteAll();
+    }
+
+    @Test
+    public void testSaveAndConfirmDuplicateNameWithDifferentId() {
+
+        //given
+        Author author = new Author("Nyor", "Ja");
+        sut.save(author);
+
+        Author author1 = author;
+        sut.save(author1);
+
+        Author author2 = new Author("Papa", "Cologne");
+        sut.save(author2);
+        sut.save(author2);
+
+        // when
+        List<Author> savedAuthors = sut.findAll();
+
+
+        // it should save 4 authors with different ids
+        assertEquals(4, savedAuthors.size());
+    }
+
+    @Test
+    public void testSaveIntegrity() {
+
+
+        Author author = new Author("daks", "oten");
+        sut.save(author);
+
+        List<Author> savedAuthors = sut.findAll();
+
+        // im having a hard time using this, im experiencing LazyInitializationException
+//        Author existingAuthor = sut.findById(author.getId());
```

Comment:
Try annotating the method with `@Transactional`

Reply:
Could you also add a `// when` comment above this line once you've fixed it?

Reply:
Hi @knjk04 ,

I try that also and still encountering an error

Reply:
Could you paste the code with the annotation here? It works for me